aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -211,7 +211,7 @@ Tensor empty_like(
     }
   }
 
-  if (self.is_quantized()) {
+  if (self.is_quantized() && options.dtype() == self.dtype()) {
```

Comment:
There is already a TORCH_CHECK later on. I think the check is better, assuming it has a good error message

Reply:
I can remove that `TORCH_CHECK`. See my comment above. Just want to add the support for the case where `self.is_quantized()` but `options.dtype() != self.dtype()`.

Reply:
actually, what about the case when self is not quantized and options.dtype is quantized? we should disallow that

Reply:
I think there is a case when options.has_dtype() is false

Reply:
Good point: It is a little mess to include all the possible logic. The basic idea is to check if we should generate the empty quantized tensor or empty normal tensor. Let me think it over about these cases.